### PR TITLE
Refactor `dftx.indexer`

### PR DESCRIPTION
**Pull request by monstrobishi**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

This parses the block sequentially per transaction, allowing us to maintain parity with a regular node's blockchain indexing.

#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

**Comment by monstrobishi** [comment]
Another advantage to this system is that we no longer have to manually keep track of the module import order of indexers i.e. having AppointOracle before SetOracleData to ensure that an oracle database entry exists to set data on.

This is now "free" thanks to transaction order and upstream validation.

**Comment by fuxingloh** [comment]
> Another advantage to this system is that we no longer have to manually keep track of the module import order of indexers i.e. having AppointOracle before SetOracleData to ensure that an oracle database entry exists to set data on.
> 
> This is now "free" thanks to transaction order and upstream validation.

This looks great!

**Comment by fuxingloh** [comment]
My bad, an earlier lint merge caused a conflict

**Comment by monstrobishi** [comment]
fixed the merge conflict @fuxingloh

**Comment by monstrobishi** [comment]
oh the dftx active price fix was merged in, let me fix that too

**Review by canonbrother — COMMENTED** [review]
yea i'm good overall

**Comment by monstrobishi** [comment]
Some funny stuff happening with the active price test after merge, will investigate first

**Review by canonbrother — DISMISSED** [review]

File: src/module.indexer/model/dftx/resign.masternode.ts [hunk 1]
```diff
@@ -59,15 +59,16 @@ export class ResignMasternodeIndexer extends DfTxIndexer<ResignMasternode> {
...
-      if (mn !== undefined) {
-        delete mn.resignTx
-        await this.masternodeMapper.put({ ...mn, resignHeight: -1 })
-      }
+  async invalidateTransaction (_: RawBlock, transaction: DfTxTransaction<ResignMasternode>): Promise<void> {
+    const data = transaction.dftx.data
+    const mn = await this.masternodeMapper.get(data.nodeId)
+    if (mn !== undefined) {
+      delete mn.resignTx
+      await this.masternodeMapper.put({ ...mn, resignHeight: -1 })
```
Comment (ivan-zynesis) [inline src/module.indexer/model/dftx/resign.masternode.ts:67]: I think this is bug before this PR, "reversed" `indexStats` needed here, increment count and TVL. Also in the invalidate createMasternode

**Review by ivan-zynesis — CHANGES_REQUESTED** [review]

File: src/module.indexer/model/dftx/resign.masternode.ts (hunk 1 above)
Comment (monstrobishi) [reply src/module.indexer/model/dftx/resign.masternode.ts:67]: It's handled in `invalidateBlockStart`, this is a special case for the masternode indexer. Trust me it works fine it's in the unit tests.

**Review by fuxingloh — APPROVED** [review]
> quite a large and necessary change

It somehow makes things more complex and more simplified at the same time.

File: src/module.indexer/model/dftx/resign.masternode.ts (hunk 1 above)
Comment (fuxingloh) [reply src/module.indexer/model/dftx/resign.masternode.ts:67]: Yeah, invalidateBlockStart reverse stats by removing the entire `indexStats` by block height

**Pull request merged**